runtime/sema/resources.go

```diff
@@ -111,19 +111,20 @@ func (ris *Resources) String() string {
 }
 
 func (ris *Resources) Get(resource interface{}) ResourceInfo {
-	return ris.resources[resource]
+	info, _ := ris.resources.Get(resource)
+	return info
 }
 
 // AddInvalidation adds the given invalidation to the set of invalidations for the given resource.
 // If the invalidation is not temporary, marks the resource to be definitely invalidated.
 //
 func (ris *Resources) AddInvalidation(resource interface{}, invalidation ResourceInvalidation) {
-	info := ris.resources[resource]
+	info, _ := ris.resources.Get(resource)
```

Comment:
can the `resource` be unavailable at this point, causing a NPD?

Reply:
Yes, and in that case `Get` will return the default value for the value type, in this case an empty `ResourceInfo`